### PR TITLE
Docs upd

### PR DESCRIPTION
**Pull request by akashkollipara**

### CODE REVIEW

File: src/driver/console/console.c [hunk 1]
```diff
@@ -15,39 +15,92 @@
...
 
+/**
+ * console_attach_device - This function links hardware driver
+ * and device driver.
+ *
+ * This function links the hardware driver and console device.
+ * On successful linking, it updates the status flag so that
+ * other functions know console is attached.
+ *
+ * @*pcon - pointer for console hardware driver
```
Comment (pranjalchanda08) [inline src/driver/console/console.c]: use annotation @param for parameter annotation. 

Ref: doxygen documentation: https://www.doxygen.nl/manual/commands.html#cmdparam 

This is to rectify and maintain the foreseen plan of using doxygen for the document generation.

File: src/driver/console/console.c [hunk 2]
```diff
@@ -15,39 +15,92 @@
...
 
-
+/**
+ * console_putc - Send char data to console device
+ *
+ * This API call writes the char data to "write" method of driver.
+ * If the console is not attached, this API returns error which
+ * needs to be handled by the caller function.
+ *
+ * @c - char (8-bits) data as input
```
Comment (pranjalchanda08) [inline src/driver/console/console.c]: @c is an annotation argument in doxygen syntax. Please use @param for taking care of such conflicts.

https://www.doxygen.nl/manual/commands.html#cmdc

File: src/driver/console/console.c [hunk 3]
```diff
@@ -15,39 +15,92 @@
...
+/**
+ * console_attach_device - This function links hardware driver
+ * and device driver.
+ *
+ * This function links the hardware driver and console device.
+ * On successful linking, it updates the status flag so that
+ * other functions know console is attached.
+ *
+ * @*pcon - pointer for console hardware driver
+ * @return - status of device/hardware driver
```
Comment (pranjalchanda08) [inline src/driver/console/console.c]: add @exception parameters to notify users about the possible exceptions or errors the function may return.

https://www.doxygen.nl/manual/commands.html#cmdexception

File: src/driver/console/console.c [hunk 4]
```diff
@@ -15,39 +15,92 @@
...
+ * If True, the console is attached and if False the console
+ * is detached.
+ */
+static bool console_attached = false;
 
+/**
+ * console_attach_device - This function links hardware driver
+ * and device driver.
+ *
+ * This function links the hardware driver and console device.
```
Comment (pranjalchanda08) [inline src/driver/console/console.c]: use @brief to write the description of any function or @fn 

https://www.doxygen.nl/manual/commands.html#cmdfn
https://www.doxygen.nl/manual/commands.html#cmdbrief

**Review by pranjalchanda08 — CHANGES_REQUESTED** [review]
Hi @akashkollipara, Please do the changes to all the files. The comments are applicable to all the files. 

Please refer: https://github.com/akashkollipara/cyancore/blob/dev_pranjalchanda08_syslog/src/driver/syslog/syslog.c

**Comment by akashkollipara** [comment]
I guess you are reviewing outdated commit....

**Review by pranjalchanda08 — APPROVED** [review]
Approving all the changes. 😃

**Pull request merged**